### PR TITLE
Bch country profile

### PR DESCRIPTION
**Pull request by tahmeedshah**
This pull request contains all files, changes for country profile including

1. Edit
2. View
3. Listing
4. Country profile card on countries page

### CODE REVIEW

File: app/views/countries/country-profile.html [hunk 1]
```diff
@@ -1,76 +1,79 @@
...
-                        <span ng-if="country.isSignatory && !country.dateSigned" class="bold">Signatory</span>
-                        <span ng-if="country.isSignatory && country.dateSigned" class="bold">Signed on {{country.dateSigned|formatDate}} </span>
-                        <span ng-if="!country.isSignatory" class="color-grey color-non-party bold">No</span>
-                    </div>
-                   
-                    <div style="white-space:nowrap;overflow-x:hidden;margin-bottom:5px;">
-                        <label>CBD Country Profile:</label>
-                        <a rel="noopener" translation-url   href="https://www.cbd.int/countries/?country={{country.code}}" target="_blank"> www.cbd.int/countries/?country={{country.code}} </a>
-                    </div>
-
```
Comment (blaisefonseca) [inline app/views/countries/country-profile.html:65]: I don't see any changes in this section still it was commited?

File: app/views/countries/country-profile.html [hunk 2]
```diff
@@ -1,76 +1,79 @@
...
-            <div country-profile code="code"></div>
+    
+        <div class="row" style="margin:0;">
+            <div class="col-xs-12" >
+                <div country-profile code="code"></div>
+            </div>
         </div>
-    </div>
-
-</div>
```
Comment (blaisefonseca) [inline app/views/countries/country-profile.html:76]: I don't see any changes in this section still it was commited?

File: app/views/forms/edit/bch/directives/edit-country-profile.directive.html [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+											<div class="km-control-group" required name="profile" caption="Profile">
+												<div km-rich-textbox required ng-model="document.profile" rows="4" placeholder="" locales="document.header.languages"></div>
+											</div>
+										</div>
+									</div> 
+
+									<div class="row">
+										<div class="col-sm-12">
+											<div class="km-control-group" name="relevantInformation" caption="Any other relevant information">
+												<div class="help-block">Please use this field to provide any other relevant information that may not have been addressed elsewhere in the record.</div>
```
Comment (blaisefonseca) [inline app/views/forms/edit/bch/directives/edit-country-profile.directive.html:48]: Please update the text as u have done in DEVDESK-417

**Review by blaisefonseca — CHANGES_REQUESTED** [review]

File: app/views/forms/edit/bch/directives/edit-country-profile.directive.js [hunk 4]
```diff
@@ -24,9 +24,65 @@ function (app, _, template) {
...
+							const identifier = result.data.response.docs[0].identifier_s;
+							if(identifier){
+								$scope.validateCountryProfileExists(identifier)
+							}
+						  }
+						
+					});
+				} 
+				$scope.checkRecordAlreadyExists();
+				$scope.validateCountryProfileExists = function(identifier){
```
Comment (blaisefonseca) [inline app/views/forms/edit/bch/directives/edit-country-profile.directive.js:48]: no need to be on scope, make this function local
`$scope.checkRecordAlreadyExists,
$scope.validateCountryProfileExists`

File: app/views/forms/edit/bch/directives/edit-country-profile.directive.js [hunk 5]
```diff
@@ -24,9 +24,65 @@ function (app, _, template) {
...
+						return;
+					}  
+					var searchQuery  = {
+						fields  : 'identifier_s',
+						query   : 'schema_s:countryProfile AND government_s:' + $scope.userGovernment().toLowerCase() 
+					}; 
+					searchService.list(searchQuery) 
+					.then(function(result) { 
+						if(result.data.response.docs.length){ 
+							const identifier = result.data.response.docs[0].identifier_s;
```
Comment (blaisefonseca) [inline app/views/forms/edit/bch/directives/edit-country-profile.directive.js:39]: const will not work in browser, chrome will handle but older browser will not

File: app/views/forms/edit/bch/edit-country-profile.js [hunk 6]
```diff
@@ -1,66 +1,9 @@
 define(['app',  './directives/edit-country-profile.directive' ], 
 function (app, _) {
-	app.controller("editCountryProfileController", ["$scope",   "$controller",   'IStorage', '$routeParams', 'ngDialog',"searchService",
+	app.controller("editCountryProfileController", ["$scope",   "$controller",
 	function($scope,  $controller,  IStorage, $routeParams, ngDialog,searchService) {
```
Comment (blaisefonseca) [inline app/views/forms/edit/bch/edit-country-profile.js:4]: `function($scope,  $controller,  IStorage, $routeParams, ngDialog,searchService) {` unused params

Comment (tahmeedshah) [reply app/views/forms/edit/bch/edit-country-profile.js:4]: IStorage removed.

File: app/views/forms/edit/bch/directives/edit-country-profile.directive.js (hunk 5 above)
Comment (tahmeedshah) [reply app/views/forms/edit/bch/directives/edit-country-profile.directive.js]: const to var

File: app/views/forms/edit/bch/directives/edit-country-profile.directive.js (hunk 4 above)
Comment (tahmeedshah) [reply app/views/forms/edit/bch/directives/edit-country-profile.directive.js]: code refactor - local functions

**Review by randyhoulahan — APPROVED** [review]

File: app/views/countries/country-profile.html (hunk 1 above)
Comment (tahmeedshah) [reply app/views/countries/country-profile.html:65]: This is due to prev commit.

File: app/views/countries/country-profile.html (hunk 2 above)
Comment (tahmeedshah) [reply app/views/countries/country-profile.html:76]: This is because of prev commit.

**Review by blaisefonseca — APPROVED** [review]

**Pull request merged**